modules/apps/staging/staging-bar-web/src/main/java/com/liferay/staging/bar/web/internal/util/StagingBarControlMenuJSPDynamicIncludeUtil.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.staging.bar.web.internal.util;
+
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.model.Layout;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.Constants;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.staging.bar.web.internal.servlet.taglib.ui.StagingBarControlMenuJSPDynamicInclude;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * @author Valmir Junior
+ */
+public class StagingBarControlMenuJSPDynamicIncludeUtil {
+
+	public static boolean isShow(HttpServletRequest httpServletRequest)
+		throws PortalException {
+
+		Boolean show = (Boolean)httpServletRequest.getAttribute(_SHOW);
+
+		if (show != null) {
+			return show;
+		}
+
+		show = _isShow(httpServletRequest);
+
+		httpServletRequest.setAttribute(_SHOW, show);
+
+		return show;
+	}
+
+	private static boolean _isShow(HttpServletRequest httpServletRequest)
+		throws PortalException {
+
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)httpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		Layout layout = themeDisplay.getLayout();
+
+		if (layout.isTypeControlPanel() || !themeDisplay.isShowStagingIcon()) {
+			return false;
+		}
+
+		String layoutMode = ParamUtil.getString(
+			httpServletRequest, "p_l_mode", Constants.VIEW);
+
+		if (layoutMode.equals(Constants.EDIT)) {
+			return false;
+		}
+
+		return true;
+	}
+
+	private static final String _SHOW =
+		StagingBarControlMenuJSPDynamicInclude.class + "#_SHOW";
```

Comment:
@Valmir-unicap This is only used to get/set an attribute from/to the `httpServletRequest`, there's no need to keep using the `StagingBarControlMenuJSPDynamicInclude` class name here.

Reply:
You can use the util's class name.